crypto/ecdsa.go

```diff
@@ -316,6 +357,15 @@ func (pk *PubKeyECDSA) Size() int {
 	return 2 * bitsToBytes((pk.goPubKey.Params().P).BitLen())
 }
 
+// given a public key (x,y), returns a compressed encoding according to X9.62 section 4.3.6
+// this compressed representation uses an extra bit to disambiguate sign
+func (pk *PubKeyECDSA) EncodeCompressed() []byte {
+	if pk.alg.curve == btcec.S256() {
+		return (*btcec.PublicKey)(pk.goPubKey).SerializeCompressed()
+	}
+	return elliptic.MarshalCompressed(pk.goPubKey, pk.goPubKey.X, pk.goPubKey.Y)
```

Comment:
It could, but https://github.com/onflow/flow-go/pull/1129/commits/d245db7c0cf99203fad6cd622fe3f1f2dc73c534 proves elliptic is not committed to multi-curve support in its ser/de functions, whereas btcec clearly will support secp256k1 indefinitely, so the choice seems obvious.

Reply:
You are right! This function [here](https://github.com/golang/go/blob/master/src/crypto/elliptic/elliptic.go#L68) is not part of the [interface](https://github.com/golang/go/blob/master/src/crypto/elliptic/elliptic.go#L27) which made btcec enable to re-use the Marshall/Unmarshall fully. Too bad, this would have led to a cleaner btcec code. 

Thanks for adding the test vector.